src/main/java/com/yelp/nrtsearch/server/handler/SearchHandler.java

```diff
@@ -271,14 +271,29 @@ public SearchResponse handle(IndexState indexState, SearchRequest searchRequest)
 
       long t0 = System.nanoTime();
 
-      hits = getHitsFromOffset(hits, searchContext.getStartHit(), searchContext.getTopHits());
+      if (searchContext.getFetchTasks().getHitsLoggerFetchTask() != null) {
```

Comment:
I wasn't too sure if I should create a new method or just modify `getHitsFromOffset` to require hitsToLog as another parameter. I decided to create a new method to make it more clear that the logic is a bit different when there are hits to log. Also, modifying the `getHitsFromOffset` would need to update its call in other two classes: `InnerHitFetchTask` and `TopHitsCollectorManager`. 

I don't have a strong opinion on which approach to choose so let me know if you prefer one over the other or if you have other suggestions.